packages/flutter/lib/src/cupertino/action_sheet.dart

```diff
@@ -0,0 +1,1264 @@
+      final Widget titleSection = new _CupertinoAlertContentSection(
+        title: title,
+        message: message,
+        scrollController: messageScrollController,
+      );
+      content.add(new Flexible(child: titleSection));
+    }
+
+    return new Container(
+      color: _kBackgroundColor,
+      child: new Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: content,
+      ),
+    );
+  }
+
+  Widget _buildActions() {
+    if (actions == null || actions.isEmpty) {
+      return new Container(
+        height: 0.0,
+      );
+    }
+    return new Container(
+      child: _CupertinoAlertActionSection(
+        children: actions,
+        scrollController: actionScrollController,
+        hasCancelButton: cancelButton != null,
+      ),
+    );
+  }
+
+  Widget _buildCancelButton() {
+    final double cancelPadding = (actions != null || message != null || title != null)
+        ? _kEdgeHorizontalPadding : 0.0;
+    return Padding(
+      padding: new EdgeInsets.only(top: cancelPadding),
+      child: new _CupertinoActionSheetCancelButton(
+        child: cancelButton,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = <Widget>[
+      new Flexible(child: ClipRRect(
+        borderRadius: BorderRadius.circular(12.0),
+        child: new BackdropFilter(
+          filter: new ImageFilter.blur(sigmaX: _kBlurAmount, sigmaY: _kBlurAmount),
+          child: new Container(
+            decoration: _kCupertinoAlertBlurOverlayDecoration,
+            child: new _CupertinoAlertRenderWidget(
+              contentSection: _buildContent(),
+              actionsSection: _buildActions(),
+            ),
+          ),
+        ),
+      ),
+      ),
+    ];
+
+    if (cancelButton != null) {
+      children.add(
+        _buildCancelButton(),
+      );
+    }
+
+    final Orientation orientation = MediaQuery.of(context).orientation;
+    double actionSheetWidth;
+    if (orientation == Orientation.portrait) {
+      actionSheetWidth = MediaQuery.of(context).size.width - (_kEdgeHorizontalPadding * 2);
+    } else {
+      actionSheetWidth = MediaQuery.of(context).size.height - (_kEdgeHorizontalPadding * 2);
+    }
+
+    return new SafeArea(
+      child: new Semantics(
+        namesRoute: true,
+        scopesRoute: true,
+        explicitChildNodes: true,
+        label: 'Alert',
+        child: new Container(
+          width: actionSheetWidth,
+          margin: const EdgeInsets.symmetric(
+            horizontal: _kEdgeHorizontalPadding,
+            vertical: _kEdgeVerticalPadding,
+          ),
+          child: new Column(
+            children: children,
+            mainAxisSize: MainAxisSize.min,
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+/// A button typically used in an [CupertinoActionSheet].
```

Comment:
an --> a
(or should this still be ActionSheet instead of CupertinoActionSheet?)

Reply:
Yes, apologies for all of the ActionSheet/CupertinoActionSheet mixups. I believe it should be CupertinoActionSheet, in keeping with the Cupertino naming convention, but I started with it being ActionSheet and forgot to make the change throughout the comments.